docs/_templates/autosummary/class_no_inherited_members.rst

```diff
@@ -0,0 +1,38 @@
+{#
+   This is very similar to the default class template, except this one is used
+   when we don't want to generate any inherited methods.
+-#}
+
+{{ objname | escape | underline }}
+
+.. currentmodule:: {{ module }}
+
+.. autoclass:: {{ objname }}
+{#-
+   Avoid having autodoc populate the class with the members we're about to
+   summarize to avoid duplication.
+#}
+   :no-members:
+   :show-inheritance:
+{#
+   Methods all get their own separate page, with their names and the first lines
+   of their docstrings tabulated.
+-#}
+{% block methods_summary %}{% set wanted_methods = (methods | reject('in', inherited_members) | reject('==', ['__init__'])) %}{% if wanted_methods %}
```

Comment:
I'm not sure this has the desired effect. Looking at the generated artifact from the CI build I see that the `__init__` method is still getting it's own page for circuit library elements. For example, with `QuantumVolume`:

![Screenshot_2021-08-05_13-43-08](https://user-images.githubusercontent.com/2447371/128396520-f531c8d1-7687-4677-9dc3-6ab3fb12dd38.png)

I'm honestly fine if we want to leave it as a separate page and change the setting for `autoclass_content` in the conf.py to just make the class page have the class docstring and leave `__init__` as a separate entry. Either way is probably fine depending on what's easier to do.

Reply:
Whoops, my bad - it's correct in `class.rst`, but in `class_no_inherited_members.rst` I had originally tried to do something like
```
methods | reject('in', inherited_members + ['__init__'])
```
to get it in one pass, but `inherited_members` is a set (as opposed to a list), and I could immediately see how to create a set in Jinja, so I just split it into its own call.  I left in the square brackets by accident.